web/permissions/permissions.go

```diff
@@ -49,46 +63,95 @@ const ContextClaims = "token_claims"
 // Extractor extracts the permission set from the context
 func Extractor(next echo.HandlerFunc) echo.HandlerFunc {
 	return func(c echo.Context) error {
-		instance := middlewares.GetInstance(c)
-		var claims permissions.Claims
-		var err error
-
-		if token := getBearerToken(c); token != "" {
-			err = crypto.ParseJWT(token, keyPicker(instance), &claims)
-		} else if token := getQueryToken(c); token != "" {
-			err = crypto.ParseJWT(token, keyPicker(instance), &claims)
-		} else {
-			// no token is provided, hopefully the next handler does not need one
-			return next(c)
-		}
 
-		if err != nil {
+		_, _, err := extract(c)
+		if err != nil && err != ErrNoToken {
+			// no token is provided, hopefully the next handler does not need one
```

Comment:
The comment is misleading, no? When no token is provided, we don't enter in the if branch